### PR TITLE
Add new resource: Project

### PR DESCRIPTION
**Pull request by 3guboff**
Details here: https://github.com/terraform-providers/terraform-provider-cloudstack/issues/62

### CODE REVIEW

**Comment by 3guboff** [comment]
Details here: [https://github.com/terraform-providers/terraform-provider-cloudstack/issues/62](url)

**Comment by svanharmelen** [comment]
@3guboff this is not going to work for a number of reasons. Maybe you can explain what you are trying to fix with this PR?

**Comment by 3guboff** [comment]
It's working, I'm tested this changes and this changes working, it's needed for issue with new resource (Project) @svanharmelen

**Comment by 3guboff** [comment]
https://travis-ci.org/terraform-providers/terraform-provider-cloudstack/builds/517141164#L493 @svanharmelen

**Comment by svanharmelen** [comment]
All the code in the `cloudstack` package is generated by the `generate` package using the output of the `listApis` call as its input. So making changes directly to the generated code isn't going to work.

**Comment by 3guboff** [comment]
Oh, sad to read that, it's mean I need remove domain/account support from resource: project :( @svanharmelen

**Comment by svanharmelen** [comment]
I'm not saying that you need to remove it, but adding it like this isn't going to work. I asked if you could explain what you are trying to fix or add, so if you share your ideas we can look at how it can be supported.

**Comment by 3guboff** [comment]
I added this code for realizing case, when we want set pair variables: account and domainid for project. By default we create project in root domain of account what we use. Setting this variables can help set up project in specific domain and set specific owner.

**Comment by svanharmelen** [comment]
I'm still not sure if I follow why this is needed for the `project` resource? Looking at that PR I only see your are using this code here: https://github.com/terraform-providers/terraform-provider-cloudstack/pull/63/files#diff-588305dbefd01cb80904f8ece4e5a5c4R106

So is this really needed here? Will the project not be found when the domain ID isn't passed here? Not sure if you are on a public Slack somewhere so we can talk a little easier about this one?

**Comment by 3guboff** [comment]
check email @svanharmelen

File: cloudstack/ProjectService.go [hunk 1]
```diff
@@ -1072,6 +1072,9 @@ func (p *UpdateProjectParams) toURLValues() url.Values {
 	if v, found := p.p["account"]; found {
 		u.Set("account", v.(string))
 	}
+	if v, found := p.p["domainid"]; found {
+		u.Set("domainid", v.(string))
+	}
```
Comment (svanharmelen) [inline cloudstack/ProjectService.go]: This is not a valid option when updating a project (see the [docs](https://cloudstack.apache.org/api/apidocs-4.12/apis/updateProject.html))

File: cloudstack/cloudstack.go [hunk 2]
```diff
@@ -419,6 +419,34 @@ func getRawValue(b json.RawMessage) (json.RawMessage, error) {
...
+				return err
+			}
+			domain = id
+		}
+
+		ps.SetDomainid(domain)
+
+		return nil
+	}
+}
```
Comment (svanharmelen) [inline cloudstack/cloudstack.go]: This code should be added to the `generate` package and then have that recreated the `cloudstack` code.

**Comment by svanharmelen** [comment]
This is fixed (added) by PR #112

**Pull request closed without merging**